src/LinearSolvers/iterativeWrapper.jl

```diff
@@ -24,6 +24,7 @@ Example:
 
 """
 type IterativeSolver<: AbstractSolver
+   AA   # matrix being solved
```

Comment:
This sounds like a good option. Just to be clear, you store the matrices in the forward param as well, right? Looking at your code, you still build the PDE in the sensmat calculations, right? Here you'd have to add an option that checks which solver is being used and if the matrix is already available and if not then rebuild it. I think this is probably the best way to go.